Add span queries (#9485)
opensearch-project/documentation-website: _query-dsl/span/span-near.md

---
layout: default
title: Span near
parent: Span queries
grand_parent: Query DSL
nav_order: 50
---

# Span near query

The `span_near` query matches spans that are near one another. You can specify how far apart the spans can be and whether they need to appear in a specific order.

For example, you can use the `span_near` query to:

> - Find terms that appear within a certain distance of each other

Reviewer: ```suggestion
- Find terms that appear within a certain distance of each other.
```

> - Match phrases where words appear in a specific order

Reviewer: ```suggestion
- Match phrases in which words appear in a specific order.
```

> - Find related concepts that appear close together in text

Reviewer: ```suggestion
- Find related concepts that appear close to each other in a document.
```

## Example

To try the examples in this section, complete the [setup steps]({{site.url}}{{site.baseurl}}/query-dsl/span/#setup).
{: .tip}

> The following query searches for any form of "sleeve" and "long" appearing next to each other, in any order:

Reviewer: ```suggestion
The following query searches for any forms of "sleeve" and "long" appearing next to each other, in any order:
```

```json
GET /clothing/_search
{
  "query": {
    "span_near": {
      "clauses": [
        {
          "span_term": {
            "description.stemmed": "sleev"
          }
        },
        {
          "span_term": {
            "description.stemmed": "long"
          }
        }
      ],
      "slop": 1,
      "in_order": false
    }
  }
}
```
{% include copy-curl.html %}

> The query matches documents 1 ("Long-sleeved...") and 2 ("...long fluttered sleeves..."). In document 1, the words are next to each other, while in document 2 they are within the specified slop distance of `1` (there is one word between them).

Reviewer: ```suggestion
The query matches documents 1 ("Long-sleeved...") and 2 ("...long fluttered sleeves..."). In document 1, the words are next to each other, while in document 2, they are within the specified slop distance of `1` (there is 1 word between them).
```
